src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,178 @@
+     * @param input task details for task type event
+     *
+     */
+    public static void addEventTask(String input){
+        try {
+            Event newEvent = new Event(Event.getEventTask(input), Event.getEventDateTime(input));
+            taskList.add(newEvent);
+            taskCounter++;
+            System.out.println("Got it. I've added this task:");
+            printTask(newEvent);
+            TaskList.printNumberOfTasksInList(taskCounter);
+        } catch(StringIndexOutOfBoundsException se){
+            System.out.println(Errors.INVALID_TASK_DETAILS_ERROR);
+        }
+    }
+    /**
+     * Add task of type deadline to taskList
+     *
+     * @param input task details for task type deadline
+     *
+     */
+    public static void addDeadlineTask(String input){
+        try {
+            Deadline newDeadline = new Deadline(Deadline.getDeadlineTask(input), Deadline.getDeadlineDate(input));
+            taskList.add(newDeadline);
+            taskCounter++;
+            System.out.println("Got it. I've added this task:");
+            printTask(newDeadline);
+            TaskList.printNumberOfTasksInList(taskCounter);
+        } catch (StringIndexOutOfBoundsException se){
+            System.out.println(INVALID_TASK_DETAILS_ERROR);
+        }
+    }
+    /**
+     * Mark a task in taskList as complete based on index
+     *
+     * @param input user input of task they want to mark as complete
+     *
+     */
+    public static void markTaskAsComplete(String input){
+        try {
+            int taskNum = getTaskNumberArgument(input);
+            Task taskToMark = taskList.get(taskNum-1);
+            taskToMark.setDone(true);
+            printTask(taskToMark);
+            System.out.println("Nice! I've marked this task as done:\n");
+        } catch (NumberFormatException nfe){
+            System.out.println(Errors.INVALID_TASK_MARK_ERROR);
+        }
+    }
+
+    /**
+     * print all tasks in arraylist
+     *
+     * @param list Arraylist of tasks to traverse and print out
+     *
+     */
+    public static void printList(ArrayList<Task> list) {
+        System.out.println("Here are the tasks in your list:");
+        for (int i = 0; i < list.size(); i++) {
+            System.out.print(i + 1 + ".");
+            printTask(list.get(i));
+        }
+        printNumberOfTasksInList(list.size());
+    }
+
+    /**
+     * Pretty print number of tasks in list based on input
+     *
+     * @param taskCounter counter to print out neatly
+     *
+     */
+    public static void printNumberOfTasksInList(int taskCounter) {
+        System.out.println("Now you have " + taskCounter + " tasks in the list.");
+    }
+
+    /**
+     * Mark a task in taskList as incomplete (using unmark command) based on index
+     *
+     * @param input user input of task they want to mark/unmark as incomplete
+     *
+     */
+    public static void markTaskAsIncomplete(String input){
+        int taskNum = getTaskNumberArgument(input);
```

Comment:
can possibly set taskNum to getTaskNumberArgument(input) - 1 instead of -1 for each one below